### ISSUE TITLE
[java] AvoidCatchingGenericException false-positive

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#1662]
PMD version 6.11.0

```java
void foo() throws Exception {
    ...
}

void bar() {
    try {
        foo();
    } catch (Exception e) {
        ...
    }
}
```

This causes an `AvoidCatchingGenericException` warning which is wrong as there is nothing else that can be written in the catch clause.

**Comment by jsotuyod** [comment]
@boris-petrov thanks for the report. Even though throwing `Exception` is a bad practice, that is already covered with [AvoidThrowingRawExceptionTypes](https://pmd.github.io/pmd-6.11.0/pmd_rules_java_design.html#avoidthrowingrawexceptiontypes), and whitelisting this here will automatically consider overrides / library methods that are outside the scope of the developer.

**Comment by boris-petrov** [comment]
Yep, exactly, I'm catching Exception from library code. Not sure what you mean by "whitelisting" but this should only not warn when any of the methods inside the `try` block `throws Exception`, otherwise it should output the error, right?

**Comment by jsotuyod** [comment]
exactly
